internal/pkg/status/status.go

```diff
@@ -14,41 +14,141 @@ limitations under the License.
 package status
 
 import (
+	"context"
 	"fmt"
 	"io"
 
 	"gopkg.in/src-d/go-git.v4/plumbing/object"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
+	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/cli-experimental/internal/pkg/client"
 	"sigs.k8s.io/cli-experimental/internal/pkg/clik8s"
 )
 
+// Condition types
+const (
+	// Level condition
+	ConditionReady   ConditionType = "Ready"
+	ConditionSettled ConditionType = "Settled"
+
+	// Terminal condition
+	ConditionFailed    ConditionType = "Failed"
+	ConditionCompleted ConditionType = "Completed"
+)
+
 // Status returns the status for rollouts
 type Status struct {
+	// DynamicClient is the client used to talk
+	// with the cluster
+	DynamicClient client.Client
+	// Out stores the output
+	Out io.Writer
+	// Resources is a list of resource configurations
 	Resources clik8s.ResourceConfigs
-	Out       io.Writer
-	Clientset *kubernetes.Clientset
-	Commit    *object.Commit
+	// Commit is a git commit object
+	Commit *object.Commit
+}
+
+// ConditionType - condition types
+type ConditionType string
+
+// Condition - generic condition
+type Condition struct {
+	Type   ConditionType
+	Status string // metav1.ConditionStatus
+	Reason string
+}
+
+// ResourceStatus - resource status
+type ResourceStatus struct {
+	Resource   *unstructured.Unstructured
+	Conditions []Condition
+	Error      error
 }
 
 // Result contains the Status Result
 type Result struct {
-	Resources clik8s.ResourceConfigs
+	Resources []ResourceStatus
+}
+
+// NewFalseCondition condition
+func NewFalseCondition(ctype ConditionType) Condition {
+	return Condition{Type: ctype, Status: "False", Reason: ""}
+}
+
+// NewCondition condition
+func NewCondition(ctype ConditionType, reason string) *Condition {
+	return &Condition{Type: ctype, Status: "True", Reason: reason}
+}
+
+// Get - simple wrapper to avoid *
+func (c *Condition) Get() Condition {
+	return *c
+}
+
+// False - set condition to false
+func (c *Condition) False() *Condition {
+	c.Status = "False"
+	return c
 }
 
-// Do executes the apply
-func (s *Status) Do() (Result, error) {
-	fmt.Fprintf(s.Out, "Doing `cli-experimental apply status`\n")
-	if s.Commit != nil {
-		fmt.Fprintf(s.Out, "Commit %s\n", s.Commit.Hash.String())
+// GetCondition - condition
+func GetCondition(cs []Condition, ct ConditionType) *Condition {
+	for i := range cs {
+		if cs[i].Type == ct {
+			return &cs[i]
+		}
+	}
+	return nil
+}
+
+// Do executes the status
+func (a *Status) Do() (Result, error) {
```

Comment:
Do we need to provide a function to check the readiness of a list of unstructures?
```
IsReady(resources []*unstructured.Unstructured) (bool, error)
```
or 
```
IsReady(resources []*unstructured.Unstructured) ([]condition, error)
```

Reply:
That would need some context like the ioWriter, DynamiClient etc. We can discuss this in a follow up PR. It is useful.